src/controller/AutoCommissioner.cpp

```diff
@@ -285,6 +285,12 @@ CHIP_ERROR AutoCommissioner::StartCommissioning(DeviceCommissioner * commissione
     mNeedsNetworkSetup =
         mCommissioneeDeviceProxy->GetSecureSession().Value()->AsSecureSession()->GetPeerAddress().GetTransportType() ==
         Transport::Type::kBle;
+    // if we have the WiFiCredentials ,we need to set up the network
+    if (mParams.GetWiFiCredentials().HasValue())
```

Comment:
But some commissioners will pass in the credentials no matter what, even if the device is being commissioned on-network, because they don't know exactly where it will be discovered.  And then we will end up messing with the network configuration while we are talking to the device over that same network, which might break commissioning in some cases.

Again, the right thing to do here is to track the method via which the commissionnee was discovered and base this boolean on that.

Reply:
Thanks bzbarsky and I will try to find a way to track the method via which the commissionnee was discovered, this PR will be closed and I will maintaine it on my own branch.